File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/fanmanagement/FanManagementUiState.kt
package com.hhp227.concafe.presentation.main.fanmanagement

import com.hhp227.concafe.domain.model.FanManagementData
import com.hhp227.concafe.domain.model.CastClaimCandidate

data class FanManagementUiState(
    val isLoading: Boolean = true,
    val errorMessage: String? = null,
    val fanManagementData: FanManagementData? = null,
    val castClaimStatus: CastClaimStatusCard? = null,
    val castClaimSheet: CastClaimSheet? = null,
    val isClaimSheetVisible: Boolean = false,
    val isAnnouncementSheetVisible: Boolean = false,
    val announcementTitle: String = "",
    val announcementBody: String = "",
    val isSendingAnnouncement: Boolean = false,
    val infoMessage: String? = null
) {
    data class CastClaimStatusCard(
        val affiliatedCafeId: String,
        val affiliatedCafeName: String,
        val headline: String,
        val body: String,
        val accent: Accent
    )

    data class CastClaimSheet(
        val affiliatedCafeId: String,
        val affiliatedCafeName: String,
        val headline: String,
        val body: String,
        val requestableCasts: List<CastClaimCandidate> = emptyList(),
        val nextCursor: String? = null,
        val canLoadMore: Boolean = false,
        val isLoadingMore: Boolean = false,
        val selectedCastId: String? = null,
        val canSubmit: Boolean = false,
        val isSubmitting: Boolean = false
    )

    enum class QuickAction(
        val title: String,
        val subtitle: String
    ) {
        WORK_SCHEDULE("출근 관리", "이번 주 스케줄을 조정합니다."),
        CAFE_DASHBOARD("프로필 연결", "내 캐스트 프로필 연결 상태를 관리합니다.")
    }

    enum class Accent {
        PENDING,
        LINKED,
        REJECTED
    }

    val isAnnouncementSubmitEnabled: Boolean
        get() = announcementTitle.trim().isNotEmpty() &&
            announcementBody.trim().isNotEmpty() &&
            !isSendingAnnouncement

    companion object {
        fun empty(): FanManagementUiState = FanManagementUiState()
    }
}

File: iosApp/iosApp/Presentation/Main/FanManagement/FanManagementUiState.swift
//
//  FanManagementUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/07.
//

import Foundation
import Shared

struct FanManagementUiState {
    var isLoading: Bool = true
    var errorMessage: String? = nil
    var fanManagementData: FanManagementData? = nil
    var castClaimStatus: CastClaimStatusCard? = nil
    var castClaimSheet: CastClaimSheet? = nil
    var isClaimSheetVisible: Bool = false
    var isAnnouncementSheetVisible: Bool = false
    var announcementTitle: String = ""
    var announcementBody: String = ""
    var isSendingAnnouncement: Bool = false
    var infoMessage: String? = nil

    struct CastClaimStatusCard: Hashable {
        let affiliatedCafeId: String
        let affiliatedCafeName: String
        let headline: String
        let body: String
        let accent: Accent
    }

    struct CastClaimSheet {
        let affiliatedCafeId: String
        let affiliatedCafeName: String
        let headline: String
        let body: String
        let requestableCasts: [CastClaimCandidate]
        let nextCursor: String?
        let canLoadMore: Bool
        let isLoadingMore: Bool
        let selectedCastId: String?
        let canSubmit: Bool
        let isSubmitting: Bool
    }

    enum QuickAction: CaseIterable, Hashable {
        case workSchedule
        case cafeDashboard

        var title: String {
            switch self {
            case .workSchedule:
                return "출근 관리"
            case .cafeDashboard:
                return "프로필 연결"
            }
        }

        var subtitle: String {
            switch self {
            case .workSchedule:
                return "이번 주 스케줄을 조정합니다."
            case .cafeDashboard:
                return "내 캐스트 프로필 연결 상태를 관리합니다."
            }
        }
    }

    enum Accent {
        case pending
        case linked
        case rejected
    }

    var isAnnouncementSubmitEnabled: Bool {
        !announcementTitle.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !announcementBody.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !isSendingAnnouncement
    }

    static let empty = FanManagementUiState(
        isLoading: true,
        errorMessage: nil,
        fanManagementData: nil,
        castClaimStatus: nil,
        castClaimSheet: nil,
        isClaimSheetVisible: false,
        isAnnouncementSheetVisible: false,
        announcementTitle: "",
        announcementBody: "",
        isSendingAnnouncement: false,
        infoMessage: nil
    )
}
